cli/azd/cmd/templates.go

```diff
@@ -316,9 +317,18 @@ func (a *templateSourceListAction) Run(ctx context.Context) (*actions.ActionResu
 
 func newTemplateSourceAddCmd() *cobra.Command {
 	return &cobra.Command{
-		Use:   "add <key>",
-		Short: fmt.Sprintf("Adds an azd template source at the specified key %s", output.WithWarningFormat("(Beta)")),
-		Args:  cobra.ExactArgs(1),
+		Use: "add <key>",
+		Short: fmt.Sprintf(
+			"Adds an azd template source at the provided key %s\n"+
+				"  Use known-keys w/o a type for:\n"+
```

Comment:
Trying my best at word smithing here, but it wasn't intuitive for me reading the description here. I tried re-writing how I would explain keys:

> Adds an azd template source with the specified key.
> The key can be any value that uniquely identifies the template source, with the exception of the following reserved, well-known key values:
> - default: Default templates.
> - awesome-azd: Templates from http://aka.ms/awesome-azd

And I wonder if it's less intuitive since we explain all the details before the command:

![image](https://github.com/user-attachments/assets/222dba83-f44d-41c7-9364-8f7b63480618)
 
So I would recommend using examples to explain `--type` and `--location`.

There are example usages on `azd template source --help` currently but not on `azd template source add --help`: 

https://github.com/Azure/azure-dev/blob/bdcbc0d8cf0910cd66caefa84997ebdf3942bdc8/cli/azd/cmd/testdata/TestUsage-azd-template-source.snap#L34-L45

Reply:
done